freedomofpress/securedrop
# Tor functional test auto generation of  instance_information.json

## Status

Ready for review


## Description of Changes

Fixes #3621 

I specifically only tested this against the `upgrade` scenario but really any staging scenario should work that includes the `app-test` role (where the logic resides).

Changes proposed in this pull request:

* Adds an tor ansible fact gathering script to yank out tor details on the application server
* Adds logic to the `app-test` role to pull back those details into a local file for functional testing in-take

## Testing

How should the reviewer test this PR?
* `molecule converge -s upgrade` (you must be on a kvm box, and if you havent run this in a while... its going to take 30min+ depending on your connection the first run. It'll get stuck at the vagrant stage while pulling 3GB+ of boxes.)
* `molecule side-effect -s upgrade` (will upgrade instances to whatever debs you have in `build` plus build the instance_information.json locally)
* `cd securedrop && ./bin/dev-shell ./bin/run-test --capture=no -v tests/functional`

while that is going ^^ since we have a current failing test scenario the best way to confirm things are working quickly is to log into the app server:
* `molecule login -s upgrade -h app`
and then tail the apache logs to make sure connections are happening
* `tail -f /var/log/apache2/journalist-access.log` (as root)


## Deployment

Any special considerations for deployment? Consider both:

Only affects CI and local dev-env.

## Checklist

### If you made changes to the server application code:


### If you made non-trivial code changes:
